config/imsim-config.yaml

```diff
@@ -125,11 +125,15 @@ image:
                 # a small numer of photons.
 
     # The objects are processed in batches.
-    # If checkpointing is turned on, then the checkpoint file will be updated after each batch.
+    # If checkpointing is turned on, then the checkpoint file will be updated after every
+    # group of checkpoint_sampling_factor batches.  This factor helps lower the overall checkpoint
+    # rate when running thousands of processes concurrently; otherwise, the contention
+    # from lots of concurrent checkpoint writes can overwhelm the file system.
     # Even if not checkpointing, using batches helps keep the memory down, since a bunch of
     # temporary data can be purged after each batch.
     # The default number of batches is 100, but you can change it if desired.
     nbatch: 100
+    checkpoint_sampling_factor: 10
```

Comment:
We can make it less than 10, but 1 seems excessive to me.  Would be nicer IMO if we could checkpoint by numbers of objects explicitly.

Reply:
Actually, I guess now that nbatch has increased to 100, that's 10 checkpoint actions, which is probably reasonable.  I think it used to be nbatch=10 IIRC.  Maybe we should make the default be to do 10 checkpoints, whatever nbatch is?  Or separately set ncheckpoint, which could default to 10?  I'm not sure what the most reasonable default is, tbh.

Reply:
I think if we could specify how many objects we should checkpoint rather than fractions, it would be nicer.  But, I guess the way it is organized it really needs to be a  multiple of the batching amount.

10 as a default seems reasonable to me.  The "right" number is going to depend on lot on how you are using it if you really are counting on the checkpoints.

Reply:
I'm mostly trying to figure out what a user would expect if the value is absent from the config file.  That's always a hard game to play.  :)